### PR TITLE
Fix conv1d and convtranspose1d asymmetric padding

### PR DESCRIPTION
**Pull request by LielinJiang**
### PR types

Bug fixes
### PR changes

APIs
### Describe

Fix conv1d and convtranspose1d asymmetric padding

### CODE REVIEW

File: python/paddle/nn/functional/conv.py [hunk 1]
```diff
@@ -733,20 +733,31 @@ def conv_transpose1d(x,
...
+            raise ValueError(
+                "output_size should be int, or list, tuple of ints")
+
+    if output_padding == 0:
+        output_padding = []
+    else:
+        output_padding = utils.convert_to_list(output_padding, 1,
+                                               'output_padding') + [0]
+
+    if len(output_padding) > 0 and output_padding[0] > stride[0]:
```
Comment (willthefrog) [inline python/paddle/nn/functional/conv.py:756]: what is the reason behind this limitation?

Comment (LielinJiang) [reply python/paddle/nn/functional/conv.py:756]: 转置卷积是卷积的逆操作，output_padding不是对output做简单的padding，而是为了对卷积的shape对应上。具体可以看下图的解释
image

Comment (willthefrog) [reply python/paddle/nn/functional/conv.py:756]: a warning should suffice, if a user wants arbitrary padding, should not stop him/her from doing it.
did torch impose such strong restriction?

Comment (LielinJiang) [reply python/paddle/nn/functional/conv.py:756]: torch也是强制报错的。paddle的这个限制其实在c++端已经也做了。。用户想任意padding他可以自己做padding。不能因为他想做任意的padding，而去混淆output_padding的作用。

Comment (willthefrog) [reply python/paddle/nn/functional/conv.py:756]: okay, I see

**Review by willthefrog — APPROVED** [review]

**Review by iclementine — APPROVED** [review]
LGTM

**Pull request merged**